commit 5e9b7cfc209b84d135993b8cb75ea383f24b2bba
Author: Justin Stitt <justinstitt@google.com>
Date:   Tue Sep 19 05:27:45 2023 +0000

    aoe: replace strncpy with strscpy
    
    `strncpy` is deprecated for use on NUL-terminated destination strings [1].
    
    `aoe_iflist` is expected to be NUL-terminated which is evident by its
    use with string apis later on like `strspn`:
    |       p = aoe_iflist + strspn(aoe_iflist, WHITESPACE);
    
    It also seems `aoe_iflist` does not need to be NUL-padded which means
    `strscpy` [2] is a suitable replacement due to the fact that it
    guarantees NUL-termination on the destination buffer while not
    unnecessarily NUL-padding.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://manpages.debian.org/testing/linux-manual-4.8/strscpy.9.en.html [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Cc: Kees Cook <keescook@chromium.org>
    Cc: Xu Panda <xu.panda@zte.com.cn>
    Cc: Yang Yang <yang.yang29@zte.com>
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/r/20230919-strncpy-drivers-block-aoe-aoenet-c-v2-1-3d5d158410e9@google.com
    Signed-off-by: Jens Axboe <axboe@kernel.dk>

diff --git a/drivers/block/aoe/aoenet.c b/drivers/block/aoe/aoenet.c
index 63773a90581d..c51ea95bc2ce 100644
--- a/drivers/block/aoe/aoenet.c
+++ b/drivers/block/aoe/aoenet.c
@@ -39,8 +39,7 @@ static struct ktstate kts;
 #ifndef MODULE
 static int __init aoe_iflist_setup(char *str)
 {
-	strncpy(aoe_iflist, str, IFLISTSZ);
-	aoe_iflist[IFLISTSZ - 1] = '\0';
+	strscpy(aoe_iflist, str, IFLISTSZ);
 	return 1;
 }